src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -566,6 +578,28 @@ GTCodeGen::generateStencilInstantiation(const StencilInstantiation* stencilInsta
 
     StencilConstructor.startBody();
 
+    // Add static asserts to check halos against extents
+    StencilConstructor.addComment("Check if extents do not exceed the halos");
+    for(int i = 0; i < numFields; ++i) {
+      if(!StencilFields[i].IsTemporary) {
+        auto max_extents = collectedExtents[StencilFields[i].Name];
+        DAWN_ASSERT_MSG(max_extents.size() % 2 == 0,
+                      "Extents always have to come in pairs (+/-)");
+        for(int dim = 0; dim < (max_extents.size() / 2); ++dim) {
+          // assert for + accesses
```

Comment:
we will need the same protection in the naive code generation as well? 
Could we have this in a method that we can reuse between naive and gridtools code gen?